sycl/source/half_type.cpp

```diff
@@ -0,0 +1,189 @@
+//==-------------- half_type.cpp --- SYCL half type ------------------------==//
+//
```

Comment:
Probably this code comes from somewhere else.
If so it would deserve some credit and context comment...

Reply:
Actually almost all this patch is done by @shileiti . @shileiti, could you please confirm from where this code comes?

Reply:
I wrote it by myself.

Reply:
Congratulation! \o/